### ISSUE TITLE
Please add OpenBsd's sendsyslog(2) support to system source

### ISSUE THREAD

**Issue by csszep** [balabit/syslog-ng#1786]
Syslog-ng is not usable as a system syslog daemon because 
OpenBSD's syslog functions after version 5.7 use sendsyslog(2) which doesn't use
/dev/log sockets any more.

https://man.openbsd.org/sendsyslog

Related discussion:

https://marc.info/?l=openbsd-misc&m=145929600516986&w=2

**Comment by Kokan** [comment]
The link in the *Related discussion* does not work, but here is a simple C application that could listen to the `newsyslog` implementation in openBSD.

```C
#include <sys/socket.h>
#include <sys/ioctl.h>
#include <sys/un.h>
#include <fcntl.h>
#include <unistd.h>
#include <stdio.h>
#include <errno.h>

#include <syslog.h>

struct NewSyslogListener
{
        int klog;
        int pair[2];
};

int newsysloglistener_init(struct NewSyslogListener *this)
{
        if (socketpair(AF_UNIX, SOCK_DGRAM, PF_UNSPEC, this->pair) == -1) {
                printf("cannot bind socket fd\n");
                return 1;
        }
        if ((this->klog = open("/dev/klog", O_RDONLY, 0)) == -1) {
                printf("cannot open /dev/klog %d\n",errno);
                return 2;
        }

        if (ioctl(this->klog, LIOCSFD, &(this->pair[1])) == -1) {
                printf("cannot ioctl klog LIOCSFD\n");
                return 3;
        }

        close(this->pair[1]);

        return 0;
};

void newsysloglistener_deinit(struct NewSyslogListener *this)
{
        close(this->pair[0]);
        close(this->klog);
}

int main()
{
        struct NewSyslogListener newsyslog;
        int res = -1;

        res = newsysloglistener_init(&newsyslog);
        if (res) {
                return res;
        }

        struct sockaddr_un sa;
        socklen_t          salen;
        ssize_t            n;
        char               linebuf[1025];

        salen = sizeof(sa);

        while (1) {
                n = recvfrom(newsyslog.pair[0], linebuf, 1024, 0, (struct sockaddr*)&sa, &salen);
                if (n>0) printf("msg: %.*s\n", (int)n, linebuf);
                else printf("nothing to print\n");
        };

        newsysloglistener_deinit(&newsyslog);


        return 0;
};
```

**Comment by csszep** [comment]
Sorry, newer discussion:

https://marc.info/?l=openbsd-ports&m=151198551018582&w=2

**Comment by Kokan** [comment]
Yes it is quite clear that in the `openBSD` they created a unique solution for logging.

Because of this, it won't work unless somebody implements this interface toward `openBSD`.
As far as I could use `google`, nobody really did (expect of course `openBSD`).

As a corner stone I created a simple `C` code, that implements (partially) that interface.
The above code shall be implemented in `syslog-ng` as a new module, and at that point it is going to be able to fetch the logs, I do not see any workaround possibility with the current `syslog-ng` configuration options.

I shared that code, because I do not know when do I have the time to implement such thing, but could not find a good description about the changes, so it can be helpful is somebody else up to the task.
